app/controllers/chats_controller.rb

```diff
@@ -0,0 +1,101 @@
+class ChatsController < ApplicationController
+	before_action :authenticate_user!
+	def index
+		all_recipients
+
+		if @notified_users.count > 0
+			@user = @notified_users.first
+		elsif @followed_users.count > 0
+			@user = @followed_users.first
+		else
+			@user = @all_users.first
+		end
+		@msgs = get_messages(@user)
+		get_read(@msgs)
+	end
+
+	def msg
+		all_recipients
+		
+		@user = User.find(params[:id])
+		@msgs = get_messages(@user)
+		get_read(@msgs)
+	end
+
+	def send_msg
+
+		@user = User.find(params[:chat][:receiver_id])
+		chat = Chat.new(chat_params)
+		chat.sender_id = current_user.id
+		chat.message_status = "unread"
```

Comment:
duplicate the params and then merge the duplicated new hash with new data instead of assigning one by one all values.
```
new_paramss = params.dup
new_params.merge!({sender_id: current_user.id, message_status: 'unread', any other params })
chat  = Chat.new(new_params)
```